### PR TITLE
Feat/review page

### PR DESCRIPTION
**Pull request by mlauren**
## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#231

## Screenshots
Screen Shot 2022-06-07 at 9 09 15 AM
Screen Shot 2022-06-07 at 9 24 27 AM

No fields selected:
Screen Shot 2022-06-07 at 9 25 00 AM
edit page: 

Screen Shot 2022-06-07 at 10 10 57 AM

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/routing/Page.tsx [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
-import { useNavigate, To, useLocation } from 'react-router-dom';
+import {
+  useNavigate,
+  To,
+  useLocation,
+  useSearchParams,
+} from 'react-router-dom';
 import { PageProps, IFormData } from './types';
 import { RouterContext } from './RouterContext';
+import { Link } from 'react-router-dom';
```
Comment (bjmccotter7192) [inline src/routing/Page.tsx:11]: You mind moving this into the `react-router-dom` import above?

File: examples/burial-form/ReviewPage.jsx [hunk 2]
```diff
@@ -0,0 +1,345 @@
+import React from 'react';
+import {
+  DebuggerView
+} from '@department-of-veterans-affairs/va-forms-system-core';
+import { useFormikContext } from 'formik';
+import { Page } from '@department-of-veterans-affairs/va-forms-system-core';
```
Comment (bjmccotter7192) [inline examples/burial-form/ReviewPage.jsx:6]: Looks like `Page` isn't being used so could you remove the import?

File: examples/burial-form/ReviewPage.jsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    return (
+      <div className={`level-${rank}-field-${key}`} key={`level-${rank}-field-${key}`}> 
+        {fieldLabel} 
+        <span className={'review-page--page-info--value-text field-value' + (rank > 0 && ` field-value-level-${rank}`)}> { transformFieldValue(key, field) }</span>
+      </div>
+    );
+  }
+}
+
+export default function ReviewPage(props) {
```
Comment (bjmccotter7192) [inline examples/burial-form/ReviewPage.jsx:52]: Looks like props isn't being used here, is the `title` prop suppose to be used somewhere?

File: examples/burial-form/ReviewPage.jsx [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+  }
+  if (field.value === 'false' | field.value === false ) {
+    return 'No';
+  }
+  if (["from", "to", "veteranDateOfBirth", "deathDate", "burialDate"].indexOf(key) > -1) {
+    const date = new Date(field.value);
+    return date.toLocaleDateString("en-US", { day: 'numeric', month: 'long', year: 'numeric' });
+  }
+  if (["amountIncurred", "amountGovtContribution"].indexOf(key) > -1) {
+    return "$" + field.value
```
Comment (bjmccotter7192) [inline examples/burial-form/ReviewPage.jsx:22]: Getting this ESLint error: `Operands of '+' operation must either be both strings or both numbers. Consider using a template literal. eslint (@typescript-eslint/restrict-plus-operands)`

File: examples/burial-form/ReviewPage.jsx [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+  const fieldLabel = field.label && <label className='vads-u-margin-top--1 review-page--page-info--label-text'>{field.label}:</label>
+
+  if ((typeof field.value) === 'object') {
+    return (<div className={`level-${rank}-field-${key}`} key={`level-${rank}-field-${key}`}> {bufferFields(field.value, rank+1)}</div>)
+  }
+  else {
+    return (
+      <div className={`level-${rank}-field-${key}`} key={`level-${rank}-field-${key}`}> 
+        {fieldLabel} 
+        <span className={'review-page--page-info--value-text field-value' + (rank > 0 && ` field-value-level-${rank}`)}> { transformFieldValue(key, field) }</span>
```
Comment (bjmccotter7192) [inline examples/burial-form/ReviewPage.jsx:46]: Also getting an ESLint rule error here: `Operands of '+' operation must either be both strings or both numbers. Consider using a template literal. eslint(@typescript-eslint/restrict-plus-operands)`

Maybe we should convert to string to be safe?

File: examples/burial-form/ReviewPage.jsx [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+            label: "Did a federal/state government or the Veteran’s employer contribute to the burial?  (Not including employer life insurance)",
+            value: state?.values?.govtContributions
+          },
+          "amountGovtContribution": {
+            label: "Amount of government or employer contribution",
+            value: state?.values?.amountGovtContribution
+          }
+        },
+      },
+      // {
```
Comment (bjmccotter7192) [inline examples/burial-form/ReviewPage.jsx:277]: Does this need to be commented out because the page isn't there to navigate back to?

**Review by bjmccotter7192 — COMMENTED** [review]
Looks good left a few comments but nothing major, good work!

**Comment by bjmccotter7192** [comment]
Soooo after testing this also looks like the navigation back and forth from the review page to the edit page seems to be working as well does this PR cover this ticket as well: https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/342

File: examples/burial-form/ReviewPage.jsx (hunk 6 above)
Comment (mlauren) [reply examples/burial-form/ReviewPage.jsx]: I can go ahead and uncomment this, so when the page is available, the data shows up in the review page.

**Pull request merged**